Mathlib/GroupTheory/MaximalSubgroups.lean

```diff
@@ -0,0 +1,67 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir
+-/
+
+import Mathlib.Algebra.Group.Subgroup.Basic
+import Mathlib.Order.Atoms
+
+/-! # Maximal subgroups
+
+A subgroup `IsMaximal` if it is maximal in the collection of
+proper subgroups.
+
+We prove a few basic results which are essentially copied from
+those about maximal ideals.
+
+Besides them, we have :
+* `isMaximal_iff` proves that a subgroup K of G is maximal
+if and only  if it is not ⊤ and if for all g ∈ G such that g ∉ K,
+every subgroup containing K and g is ⊤.
+
+## TODO
+
+Is it useful to write it in a greater generality?
+Will we need to write this for all `sub`-structures ?
+
+-/
+
+
+variable {G : Type*} [Group G]
+
+namespace Subgroup
+
+/-- A subgroup is maximal if it is maximal in the collection of proper subgroups. -/
+class IsMaximal (K : Subgroup G) : Prop where
+/-- A subgroup is maximal if it is maximal in the collection of proper subgroups. -/
+  out : IsCoatom K
+
+theorem isMaximal_def {K : Subgroup G} : K.IsMaximal ↔ IsCoatom K :=
+  ⟨fun h => h.1, fun h => ⟨h⟩⟩
+
+theorem IsMaximal.ne_top {K : Subgroup G} (h : K.IsMaximal) : K ≠ ⊤ :=
+  (isMaximal_def.1 h).1
+
+theorem isMaximal_iff {K : Subgroup G} :
+    K.IsMaximal ↔ K ≠ ⊤ ∧ ∀ (H : Subgroup G) (g), K ≤ H → g ∉ K → g ∈ H → H = ⊤ := by
```

Comment:
This should be true for any `SetLike` with `OrderTop`, if you use `IsCoatom` instead of a custom typeclass. Something like `simp [IsCoatom, SetLike.lt_iff_le_and_exists]` should work (not tested).

Reply:
I'll try.

Reply:
Before I embark in this, I would like your opinion: this is done in a similar way for maximal ideals, see https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Ideal/Maximal.html#Ideal.isMaximal_iff 
It makes no sense to make things differently, that would be out of the scope of the present PR to redefine maximal ideals.